src/range_proof/party.rs

```diff
@@ -206,6 +215,28 @@ impl<'a> PartyAwaitingBitChallenge<'a> {
     }
 }
 
+/// Overwrite secrets with null bytes when they go out of scope.
+impl<'a> Drop for PartyAwaitingBitChallenge<'a> {
+    fn drop(&mut self) {
+        self.v.clear();
+        self.v_blinding.clear();
+        self.a_blinding.clear();
+        self.s_blinding.clear();
+
+        // Important: due to how ClearOnDrop auto-implements InitializableFromZeroed
+        // for T: Default, calling .clear() on Vec compiles, but does not
+        // clear the content. Instead, it only clears the Vec's header.
+        // Clearing the underlying buffer item-by-item will do the job, but will
+        // keep the header as-is, which is fine since the header does not contain secrets.
+        for e in self.s_L.iter_mut() {
+            e.clear();
+        }
+        for e in self.s_R.iter_mut() {
+            e.clear();
+        }
```

Comment:
This is not directly possible for `Vec<Scalar>` due to

```
the trait `clear_on_drop::clear::InitializableFromZeroed` is not implemented for `[curve25519_dalek::scalar::Scalar]` 
```

But since we use vectors of scalars in 4 places already (and even more later when we merge in the `circuit` branch), maybe I should make my own wrapper that clears each element on drop?

Reply:
Or make a PR against curve25519-dalek to add `ZeroSafe` to Scalar too?

Reply:
Filed https://github.com/dalek-cryptography/curve25519-dalek/pull/202

Reply:
Hmm, I guess the reason that it works internally in curve25519-dalek is that what's actually zeroed is a buffer of digits, which is just an integer array (and therefore zero-safe), or something like taht